src/requests/process-method.ts

```diff
@@ -89,10 +89,26 @@ function getParamSeparation(paramGroups: Dictionary<Parameter[]>): string[] {
       return def;
     }
 
-    // only one direct body parameter is allowed
-    // 1st one taken if more of them present
     if (groupName === 'body') {
-      def = `params.${group[0].name};`;
+      // when the schema: { '$ref': '#/definitions/exampleDto' } construct is used
+      if ('schema' in group[0]) {
+        def = `params.${group[0].name};`;
+      } else {
+        const list = _.map(group, p => `${p.name}: params.${p.name},`);
+        def = '{\n' + indent(list) + '\n};';
+      }
+
+      // when the patch method is used, it will discard bodyParams keys with value === undefined
```

Comment:
Happens for all `body` params, not only for `patch`, does not it?

Reply:
comment changed